app/grandchallenge/reader_studies/forms.py

```diff
@@ -310,6 +338,54 @@ def interface_choices(self):
     def initial_interface(self):
         return self.interface_choices().first()
 
+    def widget_choices(self):
+        answer_type = self["answer_type"].value()
+        choices = [("", "Default")]
+        if answer_type:
+            try:
+                extra_options = ANSWER_TYPE_TO_QUESTION_WIDGET[answer_type]
+                choices.extend(
+                    (option.name, option.label) for option in extra_options
+                )
+            except KeyError:
+                pass
+        return choices
+
+    def initial_widget(self):
+        try:
+            return self.instance.widget.kind
+        except ObjectDoesNotExist:
+            return ""
+
+    def clean_widget(self):
+        value = self.cleaned_data.get("widget")
+        if value == "":
+            return None
+        else:
+            widget_instance = QuestionWidget(kind=value)
+
+            if value == QuestionWidgetKindChoices.ACCEPT_REJECT:
+                if not self.cleaned_data.get("interface"):
+                    self.add_error(
+                        error=ValidationError(
+                            f"To use the {widget_instance.get_kind_display()} widget you must define a default answer."
+                        ),
+                        field=None,
+                    )
+                if self.cleaned_data.get("required"):
+                    self.add_error(
+                        error=ValidationError(
+                            f'To use the {widget_instance.get_kind_display()} widget you must uncheck "required".'
+                        ),
+                        field=None,
+                    )
+            if not self.errors:
+                try:
+                    return self.instance.widget
```

Comment:
I don't really understand what is happening here. You're creating a new widget in memory, then validating that, but here returning the initial value? What if it is of a different kind? Is it not validated given the existing options?

Reply:
Ah good catch, this was a remnant of the previous implementation. It doesn't really matter in the current implementation since `self.instance.widget` == `widget_instance`. The type is limited by the `widget_choices` and automatically validated. But I have removed this part, since really all we want to do here is send along the new widget instance.